$NetBSD: patch-aa,v 1.25 2017/04/11 09:33:30 adam Exp $

Define LD_LIBRARY_PATH in this script as it is not propagated from outside.
Make this pkgsrc friendly.
postconf with stdin from /dev/null: it reads none, and the cross build's irix-exec bridge would forward the loop's input to it.

--- postfix-install.orig
+++ postfix-install
@@ -208,6 +208,7 @@
 IFS=" 	
 "
 BACKUP_IFS="$IFS"
+LD_LIBRARY_PATH=`pwd`/lib; export LD_LIBRARY_PATH
 
 # This script uses outputs from Postfix and non-Postfix commands.
 # Override all LC_* settings and LANG for robustness.
@@ -271,7 +272,7 @@
     case "$junk" in
     *MAIL_VERSION*) 
 	case "$mail_version" in
-	"") mail_version="`bin/postconf -dhx mail_version`" || exit 1
+	"") mail_version="`bin/postconf </dev/null -dhx mail_version`" || exit 1
 	esac
 	val=`echo "$junk" | sed 's/MAIL_VERSION$/'"$mail_version/g"` || exit 1
 	case "$val" in
@@ -459,27 +460,7 @@
 
 : ${install_root=/}
 : ${tempdir=`pwd`}
-: ${config_directory=`bin/postconf -c conf -h -d config_directory`}
-
-# Find out the location of installed configuration files.
-
-test -z "$non_interactive" && for name in install_root tempdir config_directory
-do
-    while :
-    do
-	echo
-	eval echo Please specify \$${name}_prompt | ${FMT}
-	eval echo \$n "$name: [\$$name]\  \$c"
-	read ans
-	case $ans in
-	"") break;;
-	 *) case $ans in
-	    /*) eval $name=$ans; break;;
-	     *) echo; echo $0: Error: $name should be an absolute path name. 1>&2;;
-	    esac;;
-	esac
-    done
-done
+: ${config_directory=`bin/postconf </dev/null -c conf -h -d config_directory`}
 
 # In case some systems special-case pathnames beginning with //.
 
@@ -525,7 +506,7 @@
 	case "$junk" in
 	"") eval unset $name;;
 	esac
-	eval : \${$name=\`bin/postconf -qc $CONFIG_DIRECTORY -hx $name\`} ||
+	eval : \${$name=\`bin/postconf </dev/null -qc $CONFIG_DIRECTORY -hx $name\`} ||
 	    exit 1
     done
 }
@@ -538,7 +519,7 @@
     case "$junk" in
     "") eval unset $name;;
     esac
-    eval : \${$name=\`bin/postconf -c conf -d -hx $name\`} || exit 1
+    eval : \${$name=\`bin/postconf </dev/null -c conf -d -hx $name\`} || exit 1
 done
 
 # Override settings manually.
@@ -846,7 +827,7 @@
 # the wrong place when Postfix is being upgraded.
 
 case "$mail_version" in
-"") mail_version="`bin/postconf -dhx mail_version`" || exit 1
+"") mail_version="`bin/postconf </dev/null -dhx mail_version`" || exit 1
 esac
 
 # Undo MAIL_VERSION expansion at the end of a parameter value. If
@@ -866,7 +847,7 @@
     esac
 done
 
-bin/postconf -qc $CONFIG_DIRECTORY -e \
+bin/postconf </dev/null -qc $CONFIG_DIRECTORY -e \
     "daemon_directory = $daemon_directory" \
     "data_directory = $data_directory" \
     "command_directory = $command_directory" \
@@ -883,13 +864,3 @@
     "shlib_directory = $shlib_directory" \
     "meta_directory = $meta_directory" \
 || exit 1
-
-# If Postfix is being installed locally from source code, do the
-# post-install processing now.
-
-# The unexpansion above may have side effects on exported variables.
-# It does not matter because bin/postfix below will override them.
-
-test -n "$install_root" || {
-    bin/postfix post-install $post_install_options || exit 1
-}
